src/Agent.Worker/Variables.cs

```diff
@@ -403,13 +413,26 @@ public void Set(string name, string val, bool secret = false)
                 // Store the value as-is to the expanded dictionary and the non-expanded dictionary.
                 // It is not expected that the caller needs to store an non-expanded value and then
                 // retrieve the expanded value in the same context.
-                var variable = new Variable(name, val, secret);
+                var variable = new Variable(name, val, secret, readOnly);
                 _expanded[name] = variable;
                 _nonexpanded[name] = variable;
                 _trace.Verbose($"Set '{name}' = '{val}'");
             }
         }
 
+        public bool IsReadOnly(string name)
+        {
+            if (!Read_Only_Variables)
+            {
+                return false;
+            }
+            Variable existingVariable = null;
+            if (!_expanded.TryGetValue(name, out existingVariable)) {
```

Comment:
nit: let's fix the formatting here please

```
if (!Read_Only_Variables) 
{ 
    return false; 
} 

Variable existingVariable = null; 
if (!_expanded.TryGetValue(name, out existingVariable)) 
{
     _nonexpanded.TryGetValue(name, out existingVariable); 
} 

return (existingVariable != null && IsReadOnly(existingVariable));

```

Reply:
Done